src/parser/core/EventFilter.ts

```diff
@@ -7,7 +7,7 @@ const VALID_BY_FLAGS = SELECTED_PLAYER | SELECTED_PLAYER_PET;
 export type SpellInfo = {
   id: number;
 };
-export type SpellFilter<T extends SpellInfo = SpellInfo> = T | T[];
+export type SpellFilter<T extends SpellInfo = SpellInfo> = T | readonly T[];
```

Comment:
I don't think I have any issues with this, but curious about the change.

Reply:
i'm a fan of `as const` so this just lets me pass in something like this while setting up an event listener https://github.com/WoWAnalyzer/WoWAnalyzer/pull/5722/commits/04c5418bda8e7bdf92868d301b42420b6114812e#diff-772fdc75020e2577a67c4499412077cb9a6706b8780828867e8426f2ce2047b8R13-R23